buildSrc/src/main/groovy/airbyte-bulk-connector.gradle

```diff
@@ -156,14 +171,32 @@ class AirbyteBulkConnectorPlugin implements Plugin<Project> {
             dependsOn project.tasks.matching { it.name ==~ /(compile|spotbugs)[a-zA-Z]*Java/ }
         }
 
-        project.tasks.register('integrationTestJava', Test) {
-            description = 'Runs the integration tests.'
+        // This task exists solely so that we can run the "modern" integration tests
+        // in non-docker mode on our laptops.
+        // It's not a dependency of any other task.
+        project.tasks.register('integrationTestNonDocker', Test) {
+            description = 'Runs the integration tests in non-docker mode.'
             group = 'verification'
-            testClassesDirs = project.sourceSets.integrationTestJava.output.classesDirs
-            classpath = project.sourceSets.integrationTestJava.runtimeClasspath
+            testClassesDirs = project.sourceSets.integrationTest.output.classesDirs
+            classpath = project.sourceSets.integrationTest.runtimeClasspath
             useJUnitPlatform()
         }
 
+        // For historical reasons (i.e. airbyte-ci), this task is called integrationTestJava.
+        // airbyte-ci only wants to run a task with exactly that name.
+        // There's intentionally no equivalent to run integrationTestNonDocker + integrationTestLegacy.
+        // It's an actual Test task (rather than just depending on two sub-tasks integrationTestDocker + integrationTestLegacy)
+        // so that gradle args (e.g. `--tests`) work as expected.
+        project.tasks.register('integrationTestJava', Test) {
+            description = 'Runs the integration tests in docker mode.'
+            group = 'verification'
+            testClassesDirs = project.sourceSets.integrationTest.output.classesDirs + project.sourceSets.integrationTestLegacy.output.classesDirs
+            classpath = project.sourceSets.integrationTest.runtimeClasspath + project.sourceSets.integrationTestLegacy.runtimeClasspath
+            useJUnitPlatform()
+            // We need a docker image to run this task, so depend on assemble
+            dependsOn project.tasks.assemble
+            environment "MICRONAUT_ENVIRONMENTS", "DOCKERIZED_INTEGRATION_TEST"
+        }
```

Comment:
Don't bother doing this if it's too much work but it would be nice if these `integrationTest` sources sets and tasks were only added on destinations connectors. The plugin should be able to deduce whether the connector it's being applied to is a destination or not based on the gradle project name.

Reply:
done https://github.com/airbytehq/airbyte/pull/46691/commits/4d7990a7dd521b965ac259c87549b047f256f537